CONTRIBUTING.md

```diff
@@ -0,0 +1,166 @@
+
+#### Repository secrets and variables that required configuring
+
+##### Repository variables
+
+To set a repository variable:
+1. Navigate to your repository settings.
+1. Select 'Secrets and variables', then 'Actions' from the menu.
+1. Click the 'Variables' tab.
+1. Select 'New repository variable'.
+1. Enter the variable name and value, then click 'Add variable'.
+
+**Required repository variables:**
+
+1. WRITE_GITHUB_PACKAGES_USERNAME: This requires the GitHub username you want to use to log into GitHub Container Registry.
+
+##### Repository secrets
+
+To set a repository variable:
+1. Navigate to your repository settings.
+1. Select 'Secrets and variables', then 'Actions' from the menu.
+1. Click the 'Secrets' tab.
+1. Select 'New repository secret'.
+1. Enter the secret name and value, then click 'Add secret'. Once you have done this, you will not be able to view the secret value again.
+
+###### Required repository secrets:
+
+1. GPG_KEY: This requires your Base 64 encoded GPG key payload.
+1. GPG_KEYID: This requires the ID of your GPG key in plain text.
+1. GPG_PASSPHRASE: This requires your passphrase for your GPG key in plain text.
+1. WRITE_GITHUB_PACKAGES_TOKEN: This requires your GitHub Personal Access Token with write:packages scope that you want to use to log into GitHub Container Registry.
+
+For the GPG related secrets, you can either use an existing key, or create a new key.
+
+###### How to create a new GPG key:
+1. Generate a new GPG key with `gpg --full-generate-key`
+2. Select the type of key using the following options:
+    * RSA and RSA (default)
+    * 4096 bits long
+    * Select an expiration date
+3. Enter your User ID for the key:
+    * Name: Your full name
+    * Email: Your email address
+    * Comment: N/A
+4. Enter a password, and ensure to remember it or note it down, as you will need it later.
+
+###### How to set the GPG_KEYID repository secret: 
+
+1. Get your GPG key information with `gpg --list-secret-keys --keyid-format=short`. The output will contain something like the block below. In this example, your GPG key ID is `XXXXXXXX`. 
+```
+sec   rsa4096/XXXXXXXX 2023-05-22 [SC] [expires: 2025-05-21]
+      123456789101112131415161718192021222324252627282930
+```
+2. Create a repository secret called `GPG_KEYID` and add the value of the GPG key ID in plain text.
+
+
+###### How to set the GPG_KEY repository secret: 
+
+1. Use your GPG key ID from above to get your GPG key payload in a Base 64 encoded format with `gpg --export-secret-keys XXXXXXXX | base64`. Ensure it is on one line.
+1. Add the output to your repository secrets for GPG_KEY.
+
+###### How to set the GPG_PASSPHRASE repository secret: 
```

Comment:
Move this as the first step, so they don't have to remember the passphrase for so long?

Reply:
done